src/Numerics/Interpolation/IInterpolation.cs

```diff
@@ -65,6 +65,13 @@ public interface IInterpolation
         /// <returns>Interpolated second derivative at point t.</returns>
         double Differentiate2(double t);
 
+        /// <summary>
+        /// Differentiate three times at point t.
+        /// </summary>
+        /// <param name="t">Point t to interpolate at.</param>
+        /// <returns>Interpolated third derivative at point t.</returns>
+        double Differentiate3(double t);
```

Comment:
Adding a method to IInterpolation is a breaking change since IInterpolation is a public interface that consumers of the library can and do implement.
Might be better to leave it out of the interface but keep the method for the classes  for which the method is supported.